lib/Sema/CSGen.cpp

```diff
@@ -1330,10 +1330,17 @@ namespace {
       }
       auto *constr = cast<ConstructorDecl>(constrs.front());
       auto constrParamType = tc.getObjectLiteralParameterType(expr, constr);
+
+      // Extract the arguments.
+      SmallVector<AnyFunctionType::Param, 8> args;
+      AnyFunctionType::decomposeInput(CS.getType(expr->getArg()), args);
+
+      // Extract the parameters.
+      SmallVector<AnyFunctionType::Param, 8> params;
+      AnyFunctionType::decomposeInput(constrParamType, params);
```

Comment:
Maybe replace `getObjectLiteralParameterType` with version that returns parameters directly instead?

Reply:
One of us needs to do another pass to remove unnecessary calls to composeInput() and decomposeInput(), but I wanted to limit the scope of this PR to matchCallArguments() and immediate callers.

Reply:
Fair enough :)